polyfill/lib/duration.mjs

```diff
@@ -26,31 +26,58 @@ export class Duration {
     seconds = 0,
     milliseconds = 0,
     microseconds = 0,
-    nanoseconds = 0
+    nanoseconds = 0,
+    disambiguation = 'constrain'
   ) {
-    let tdays;
-    ({
-      days: tdays,
-      hour: hours,
-      minute: minutes,
-      second: seconds,
-      millisecond: milliseconds,
-      microsecond: microseconds,
-      nanosecond: nanoseconds
-    } = ES.BalanceTime(
-      ES.AssertPositiveInteger(ES.ToInteger(hours)),
-      ES.AssertPositiveInteger(ES.ToInteger(minutes)),
-      ES.AssertPositiveInteger(ES.ToInteger(seconds)),
-      ES.AssertPositiveInteger(ES.ToInteger(milliseconds)),
-      ES.AssertPositiveInteger(ES.ToInteger(microseconds)),
-      ES.AssertPositiveInteger(ES.ToInteger(nanoseconds))
-    ));
-    days += tdays;
+    years = ES.ToInteger(years);
+    months = ES.ToInteger(months);
+    days = ES.ToInteger(days);
+    hours = ES.ToInteger(hours);
+    minutes = ES.ToInteger(minutes);
+    seconds = ES.ToInteger(seconds);
+    milliseconds = ES.ToInteger(milliseconds);
+    microseconds = ES.ToInteger(microseconds);
+    nanonseconds = ES.ToInteger(nanonseconds);
+
+    switch (disambiguation) {
+      case 'reject':
+        for (const prop of [years, months, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds]) {
+          if (prop < 0) throw new RangeError('negative values not allowed as duration fields');
+        }
+        break;
+      case 'constrain':
+        for (let prop of [years, months, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds]) {
+          if (prop < 0) prop = -prop;
```

Comment:
I haven't got a good understanding of the various disambiguation values yes, but is this the right thing? I thought this would clamp to [0, ...].

Reply:
In https://github.com/tc39/proposal-temporal/pull/172#pullrequestreview-300605302, @pipobscure said:

> constrain => negative numbers are Math.abs but no balancing